pkg/proxy/nftables/proxier.go

```diff
@@ -1015,6 +1038,94 @@ func isAffinitySetName(set string) bool {
 	return strings.HasPrefix(set, servicePortEndpointAffinityNamePrefix)
 }
 
+// nftElementStorage is an internal representation of nftables map with single string value or set.
+type nftElementStorage struct {
+	elements      map[string]string
+	leftoverKeys  sets.Set[string]
+	containerType string
+	containerName string
+}
+
+func (c *nftElementStorage) joinKey(k []string) string {
+	return strings.Join(k, " . ")
+}
+
+func (c *nftElementStorage) splitKey(k string) []string {
+	return strings.Split(k, " . ")
+}
+
+// newNFTElementStorage creates an empty nftElementStorage and flushes the corresponding nftables container.
+func newNFTElementStorage(containerType, containerName string, tx *knftables.Transaction) *nftElementStorage {
+	c := &nftElementStorage{
+		elements:      make(map[string]string),
+		leftoverKeys:  sets.New[string](),
+		containerType: containerType,
+		containerName: containerName,
+	}
+	if containerType == "set" {
+		tx.Flush(&knftables.Set{
+			Name: containerName,
+		})
+	} else {
+		tx.Flush(&knftables.Map{
+			Name: containerName,
+		})
+	}
+	return c
+}
+
+// resetLeftoverKeys should be called before every transaction to ensure cleanupLeftoverKeys
+// will have correct data.
+func (c *nftElementStorage) resetLeftoverKeys() {
+	c.leftoverKeys = sets.New[string]()
+	for key := range c.elements {
+		c.leftoverKeys.Insert(key)
+	}
+}
+
+func (c *nftElementStorage) getValue(v []string) string {
+	if c.containerType == "set" || len(v) == 0 {
+		return ""
+	} else {
+		return v[0]
+	}
+}
```

Comment:
I don't understand this method

Reply:
will add a comment, lmk if it makes sense